kura/org.eclipse.kura.asset.provider/src/main/java/org/eclipse/kura/asset/provider/BaseAsset.java

```diff
@@ -251,8 +265,13 @@ public AssetConfiguration getAssetConfiguration() {
         return this.assetConfiguration;
     }
 
-    public void setDriver(Driver driver) {
+    public synchronized void setDriver(Driver driver) {
         this.driver = driver;
+        if (driver != null) {
+            List<DriverRecord> readRecords = getAllReadRecords();
+            hasReadChannels = !readRecords.isEmpty();
+            tryPrepareRead(readRecords);
```

Comment:
I've made a small improvement here: tryPreparedRead() is no longer called in the updated() method but only in setDriver(), because of the fact that we are re-attaching the driver every time updated() is called and after the AssetConfiguration is refreshed.
This should guarantee that we have the channel config when setDriver() is called. I've made a couple of tests and everything seems to be working.